src/SpendPermissionManager.sol

```diff
@@ -104,6 +104,9 @@ contract SpendPermissionManager is EIP712 {
     /// @notice ERC-721 interface ID (https://eips.ethereum.org/EIPS/eip-721)
     bytes4 public constant ERC721_INTERFACE_ID = 0x80ac58cd;
 
+    /// @notice Number of upper bits from spend permission hash to use in withdraw request nonce
+    uint256 public constant NONCE_HASH_BITS = 96; // Use 96 bits, leaving 160 bits in nonce for entropy
```

Comment:
Curious how you landed on this split. My intuition was flipped where we benefit from more bits for collision resistance than we need for randomness within a given permission

Reply:
per slack discussion, we're gonna go 50/50 at 128/128 bits hash/entropy